### PR TITLE
Plugins #2.1: Add name to registration request, unify plugin manager API and plugin API

### PR DESCRIPTION
**Pull request by onobrod**
## Motivation and Context
https://github.com/networkservicemesh/networkservicemesh/issues/1339

## How Has This Been Tested?

- [x] Covered by existing integration testing
- [ ] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: controlplane/pkg/apis/plugins/connection_plugin.proto [hunk 1]
```diff
@@ -2,7 +2,15 @@ syntax = "proto3";
 
 package plugins;
 
-import "github.com/networkservicemesh/networkservicemesh/controlplane/pkg/apis/connectioncontext/connectioncontext.proto";
```
Comment (haiodo) [inline controlplane/pkg/apis/plugins/connection_plugin.proto]: May be we name it just Connection?

Comment (onobrod) [reply controlplane/pkg/apis/plugins/connection_plugin.proto]: I also don't like ConnectionInfo naming, but it we name it Connection, we will get methods like `NewConnection(connection)`, `connection.getConnection()` and `connection.SetConnection(connection)`. It confuses me more. What about ConnectionWrapper?

Comment (onobrod) [reply controlplane/pkg/apis/plugins/connection_plugin.proto]: Renamed to ConnectionWrapper

Comment (haiodo) [reply controlplane/pkg/apis/plugins/connection_plugin.proto]: Protobuf generate GetLocalConnection(), GetRemoteConnection(), so I think it is ok to have it named as just a connection.

Comment (haiodo) [reply controlplane/pkg/apis/plugins/connection_plugin.proto]: We could reused it in NSM since we have interface to handle both of them.

Comment (onobrod) [reply controlplane/pkg/apis/plugins/connection_plugin.proto]: Created https://github.com/networkservicemesh/networkservicemesh/issues/1497

**Pull request merged**